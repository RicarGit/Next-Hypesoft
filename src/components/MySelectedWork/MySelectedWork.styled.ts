import styled from "styled-components"

export const MySelectedWorkSection = styled.section`
  width: 100%;
  margin-top: 180px;

  @media (max-width: 450px) {
    padding: 10px;
    margin-top: 100px;
  }
`

export const WorkCardContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 24px;
`