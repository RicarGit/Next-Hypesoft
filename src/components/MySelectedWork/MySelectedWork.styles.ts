import styled from "styled-components"

export const MySelectedWorkSection = styled.section`
  width: 100%;
  margin-top: 181px;
  display: flex;
  align-items: center;
`

export const Container = styled.div`
  width: 1320px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const WorkCardContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 24px;
`