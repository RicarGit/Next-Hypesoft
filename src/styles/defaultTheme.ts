import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme