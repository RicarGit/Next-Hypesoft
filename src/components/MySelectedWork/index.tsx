import * as S from './styles'

import selectedWorkImage1 from './selected_work_image1.svg'
import selectedWorkImage2 from './selected_work_image2.svg'
import selectedWorkImage3 from './selected_work_image3.svg'
import selectedWorkImage4 from './selected_work_image4.svg'
import selectedWorkImage5 from './selected_work_image5.svg'
import selectedWorkImage6 from './selected_work_image6.svg'

import SectionHeader from 'shared/SectionHeader'
import WorkCard from 'components/WorkCard'

const MySelectedWork = () => {
  return (
    <S.MySelectedWorkSection id='works'>
      <S.Container>
        <SectionHeader>My Selected Work</SectionHeader>
        <S.WorkCardContainer>

          <WorkCard workImage={selectedWorkImage1} alt='work 1' >
            Eduguard - E-learning Website
          </WorkCard>

          <WorkCard workImage={selectedWorkImage2} alt='work 2' className='active' >
            Eduguard - E-learning Website
          </WorkCard>

          <WorkCard workImage={selectedWorkImage3} alt='work 3' >
            Huma Marketing Agency
          </WorkCard>

          <WorkCard workImage={selectedWorkImage4} alt='work 4' >
            Beeryblog Blog Website
          </WorkCard>

          <WorkCard workImage={selectedWorkImage5} alt='work 5' >
            Educare Elearning website
          </WorkCard>

          <WorkCard workImage={selectedWorkImage6} alt='work 6' >
            Blogy Blog Website
          </WorkCard>

        </S.WorkCardContainer>
      </S.Container>
    </S.MySelectedWorkSection>
  )
}

export default MySelectedWork