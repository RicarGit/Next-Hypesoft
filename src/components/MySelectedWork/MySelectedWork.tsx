import * as S from './MySelectedWork.styles'

import { Assets } from '~/assets'

import { SectionsTitle } from '~/components/shared/SectionsTitle'
import { WorkCard } from '~/components/WorkCard'

export const MySelectedWork = () => {
  return (
    <S.MySelectedWorkSection id='works'>
      <S.Container>
        <SectionsTitle>My Selected Work</SectionsTitle>

        <S.WorkCardContainer>
          <WorkCard workImage={Assets.MyWorksImage1} alt='work 1' >
            Eduguard - E-learning Website
          </WorkCard>

          <WorkCard workImage={Assets.MyWorksImage2} alt='work 2' className='active' >
            Eduguard - E-learning Website
          </WorkCard>

          <WorkCard workImage={Assets.MyWorksImage3} alt='work 3' >
            Huma Marketing Agency
          </WorkCard>

          <WorkCard workImage={Assets.MyWorksImage4} alt='work 4' >
            Beeryblog Blog Website
          </WorkCard>

          <WorkCard workImage={Assets.MyWorksImage5} alt='work 5' >
            Educare Elearning website
          </WorkCard>

          <WorkCard workImage={Assets.MyWorksImage6} alt='work 6' >
            Blogy Blog Website
          </WorkCard>
        </S.WorkCardContainer>
      </S.Container>
    </S.MySelectedWorkSection>
  )
}