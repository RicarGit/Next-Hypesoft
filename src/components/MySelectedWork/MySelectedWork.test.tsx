import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import MySelectedWork from '.'

describe('MySlectedWork', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <MySelectedWork />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})