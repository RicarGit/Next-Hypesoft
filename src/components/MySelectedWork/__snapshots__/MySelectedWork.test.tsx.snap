// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySlectedWork should match snapshot 1`] = `
<section
  class="sc-bczRLJ cSByKk"
  id="works"
>
  <div
    class="sc-gsnTZi knAhub"
  >
    <div
      class="sc-hKMtZM iUPNYd"
    >
      <h2
        class="sc-eCYdqJ kwaGpO"
      >
        My Selected Work
      </h2>
    </div>
    <div
      class="sc-dkzDqf iaAiWh"
    >
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 1"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Eduguard - E-learning Website
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 2"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Eduguard - E-learning Website
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            View Project
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 3"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Huma Marketing Agency
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 4"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Beeryblog Blog Website
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 5"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Educare Elearning website
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-jSMfEi bRxKxV"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="work 6"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="sc-gKXOVf ewNPvu"
        >
          <h3
            class="sc-iBkjds delcgH"
          >
            Blogy Blog Website
          </h3>
          <div
            class="sc-ftvSup dACrzl"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="pink arrow"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
